src/settings-ui/Settings.UI/Styles/TextBlock.xaml

```diff
@@ -20,4 +20,11 @@
         <Setter Property="FontSize" Value="{StaticResource SecondaryTextFontSize}"/>
         <Setter Property="Foreground" Value="{ThemeResource TextFillColorSecondaryBrush}"/>
     </Style>
+
+    <x:Double x:Key="HeaderTextFontSize">12</x:Double>
```

Comment:
You can use {StaticResource BodyStrongTextBlockStyle} instead of creating a new style. BodyStrongTextBlockStyle comes with WinUI :)